r/R/filesystem.R

```diff
@@ -273,33 +273,53 @@ FileSystem$from_uri <- function(uri) {
   fs___FileSystemFromUri(uri)
 }
 
-get_path_and_filesystem <- function(x, filesystem = NULL) {
+get_paths_and_filesystem <- function(x, filesystem = NULL) {
   # Wrapper around FileSystem$from_uri that handles local paths
   # and an optional explicit filesystem
   if (inherits(x, "SubTreeFileSystem")) {
```

Comment:
This check seems to assume a singular x

Reply:
In the multiple files case, `x` is always a character vector. This function was also written to handle the case in which `x` is a `SubTreeFileSystem`, and I preserved that with no changes.

Reply:
(because remember this plural variant also serves the singular case; `get_path_and_filesystem()` calls it)